CMakeLists.txt

```diff
@@ -1587,6 +1572,21 @@ if(BUILD_TESTING)
   add_subdirectory(tests)
 endif()
 
+#libpsl
+option(CURL_USE_LIBPSL "Use libPSL" ON)
+mark_as_advanced(CURL_USE_LIBPSL)
+set(USE_LIBPSL OFF)
+
+if(CURL_USE_LIBPSL)
+  find_package(LibPSL)
+  if(LIBPSL_FOUND)
+    list(APPEND CURL_LIBS ${LIBPSL_LIBRARY})
+    list(APPEND CMAKE_REQUIRED_INCLUDES "${LIBPSL_INCLUDE_DIR}")
+    include_directories("${LIBPSL_INCLUDE_DIR}")
+    set(USE_LIBPSL ON)
+  endif()
+endif()
+
```

Comment:
Doesn't this effectively disables libpsl in the build?

Detecting and enabling at this position sets `USE_LIBPSL` after it is used in `add_subdirectory(lib)`.

Reply:
I will check. Thank you for this note.

Reply:
Yes you are right. I check with:
```
curl --version
curl 8.7.0-DEV (Windows) libcurl/8.7.0-DEV Schannel zlib/1.3.1
Release-Date: [unreleased]
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS HSTS HTTPS-proxy IPv6 Kerberos Largefile libz NTLM SPNEGO SSL SSPI threadsafe Unicode UnixSockets
```

Reply:
Where should I put it?
Before add_subdirectory(lib)?

Reply:
Just before the `add_subdirectory(lib)` Still I get this error.
So I don't know how to fix this errors. I will convert the fix in vcpkg side.
Should I leave this PR open?

Reply:
Are you sure it isn't a mismatched `declspec()` issue?

In curl-for-win I didn't have this problem with cmake and libpsl, but it depends on how libpsl was compiled. E.g. the vcpkg build may require to set `-DPSL_STATIC` when linking it static mode, but this is just a shot in the dark.

Reply:
ok. First I will convert my fix, then I will try your suggestion. Thank you.

Reply:
Thank you for your suggestion. I try it locally without success. I will take it to vcpkg side.